### PR TITLE
test discovery enhancements

### PR DESCRIPTION
**Pull request by ittaiz**
test discovery filters classes which match affix without tests, includes classes with @RunWith and test methods in hierarchy

### CODE REVIEW

**Comment by ittaiz** [comment]
@or-shachar I'd appreciate your review

File: src/java/io/bazel/rulesscala/test_discovery/DiscoveredTestSuite.scala [hunk 1]
```diff
@@ -134,4 +138,27 @@ object PrefixSuffixTestDiscoveringSuite {
...
+    }
+
+  private def hasTestAnnotatedMethodsInCurrentClass(testClass: Class[_]): Boolean =
+    testClass.getDeclaredMethods.exists { method =>
+      method.getAnnotations.exists { annotation: Annotation =>
+        testAnnotation.isAssignableFrom(annotation.annotationType)
+      }
+    }
+
+  private lazy val runWithAnnotation = classOf[RunWith]
```
Comment (johnynek) [inline src/java/io/bazel/rulesscala/test_discovery/DiscoveredTestSuite.scala]: why are these lazy? Seems like `val` is fine to me.

Comment (ittaiz) [inline src/java/io/bazel/rulesscala/test_discovery/DiscoveredTestSuite.scala]: We only need it if we get to that phase. I had a different thought when I started but now it's probably just an over optimization. I'll drop it.

**Comment by johnynek** [comment]
👍

**Pull request merged**